### DISCUSSION TITLE
Resetting Password for Admin-Created User Accounts via Email.

### DISCUSSION THREAD

**Discussion by yuvrajtimalsina** [filamentphp/filament#8219] [Help] [answered]
### How can an admin send a password reset link to a user after creating their account via mail?

```
  protected function afterCreate(): void
    {
        Password::sendResetLink(['email' => $this->record->email]);
    }
```
> Route [password.reset] not defined.

**I also created route and redirected user with token to reset page. It was a success until changing the password resulted :**

> Unable to find component: [filament.pages.auth.password-reset.reset-password]

I was hoping the same password reset mechanism of filamentphp to work here as well.

**Comment by leandrocfe** [comment, marked as answer]
Can you try this please?
```php
protected function afterCreate(): void
{
    $user = $this->record;
    $token = app('auth.password.broker')->createToken($user);
    $notification = new \Filament\Notifications\Auth\ResetPassword($token);
    $notification->url = \Filament\Facades\Filament::getResetPasswordUrl($token, $user);
    $user->notify($notification);
}
```
add `$panel->passwordReset()` in the PanelProvider

**Reply by yuvrajtimalsina** [reply]
Thanks @leandrocfe, It worked like a charm.

**Reply by rmirandadev** [reply]
Excelente dica meu amigo, mas tenho um porém, no caso, eu tenho dois painéis, um Admin, e um Manager, e no caso, eu crio o usuário pela Admin, e ele manda a rota para esse painel, e não o Manager. Existe algum parametro adicional que eu possa definir qual o id do painel que ele deve mandar no link para resetar a senha?

**Reply by leandrocfe** [reply]
Não tenho certeza, mas vc pode testar algum método do helper filament() ou Panel::getCurrentPanel(), algo assim. Tô no celular, não consigo checar

**Reply by pathros** [reply]
Nice it works. 
1) How do you translate the password reset form into another language?
2) After resetting the password, how do you pass the email into the input of the login page (when coming from the password reset form)?

**Comment by rmirandadev** [comment]
Solução perfeita:

`$notification->url = filament()->getPanel('editor')->getResetPasswordUrl($token, $user);`

Grato meu amigo

**Reply by asolera** [reply]
Solução perfeita para quando tem múltiplos painéis, dessa forma é possível controlar qual painel o usuário será redirecionado ao redefinir a senha, no método `getPanel()`. Show de bola, valeu!

**Comment by Muchwat** [comment]
In my User model, this works:

    public function requestResetPassword()
    {
        $status = Password::broker(Filament::getAuthPasswordBroker())->sendResetLink(
            ['email' => $this->email],
            function (CanResetPasswordContract $user, string $token) {
                $notification = new ResetPasswordNotification($token);
                 $panelName = 'admin'; // put your panel name
                 $notification->url = Filament::getPanel($panelName)->getResetPasswordUrl($token, $user);
                $user->notify($notification);
            },
        );

        if ($status !== Password::RESET_LINK_SENT) {
            Notification::make()
                ->title(__($status))
                ->danger()
                ->send();

            return;
        }

        Notification::make()
            ->title(__($status))
            ->success()
            ->send();
    }

**Comment by MirkoDzudzarLF** [comment]
If you are not creating CreateUser page and cannot use afterCreate() method, for your list page (ListUsers) for users getHeaderActions() method sidebar form you can use after() method as well.

```
Actions\CreateAction::make()
    ->slideOver()
    ->after(function (User $user) {
        $token = app('auth.password.broker')->createToken($user);
        $notification = new \Filament\Notifications\Auth\ResetPassword($token);
        $notification->url = \Filament\Facades\Filament::getResetPasswordUrl($token, $user);
        $user->notify($notification);
    }),
```
